### DISCUSSION TITLE
新增数据源插件必须要改 spi模块中的 枚举类DbType吗,另外UI也需要改吗

### DISCUSSION THREAD

**Discussion by zongkx** [apache/dolphinscheduler#11257] [Q&A] [answered]
枚举作为插件spi不够友好

**Comment by dandan0203** [comment]
新增数据源如何做，有教程吗

**Reply by DarkAssassinator** [reply]
> 新增数据源如何做，有教程吗

now there are no any detail guide, may u can refer this PR: https://github.com/apache/dolphinscheduler/pull/11157

**Comment by DarkAssassinator** [comment]
yes, now u need change UI and back-end together

**Comment by fuchanghai** [comment, marked as answer]
> 枚举作为插件spi不够友好

It seems that the data source on the UI is not the query backend
<img width="589" alt="image" src="https://user-images.githubusercontent.com/33984497/192932146-0830e266-1346-45d1-8239-c971df8bbad1.png">
